code/modules/research/nanites/nanite_programs/utility.dm

```diff
@@ -32,7 +32,12 @@
 	rogue_types = list(/datum/nanite_program/toxic)
 
 /datum/nanite_program/monitoring/enable_passive_effect()
+
+	if(!iscarbon(host_mob))
+		return
+
 	. = ..()
```

Comment:
Call ..() before checking

Reply:
..() just sets some active flag to true, which isn't needed here

Reply:
but the active flag sets it from being activated again

Reply:
The active flag makes it so each time nanites process, it calls `enable_passive_effect()`, only for it to be checked and not do anything:
`
if(check_conditions() && consume_nanites(use_rate))
		if(!passive_enabled)
			enable_passive_effect()
		active_effect()
`
isn't it better if it's never set thus never entering the `enable_passive_proc()` ever again?

Reply:
Nah I'm just dumb, didn't see the `!` in the if statement.